### DISCUSSION TITLE
Where is the codegen template?

### DISCUSSION THREAD

**Discussion by lostinplace** [nxrighthere/UnrealCLR#11] [General]
I'd like to help out, but it seems like everything is pretty dependant on [Codegen.cs](https://github.com/nxrighthere/UnrealCLR/blob/master/Source/Managed/Framework/Codegen.cs) which looks like it was itself automatically generated.  Where are the templates that were used to generate this file?

**Comment by nxrighthere** [comment]
Yes, this file is automatically generated from the C++ header using an external tool. I can't disclose it, unfortunately, but if you want to contribute, the best way to do that is to implement C++ functions ready for exposure, so I'll be able to generate C# code after merging.

See [this](https://github.com/nxrighthere/UnrealCLR/commit/5373ad03892711eade304ed55fd181df89a45b9b) commit for example: `UnrealCLR.cpp` storing a pointer to a function, `UnrealCLRFramework.cpp` implementation, `UnrealCLRFramework.h` a prototype.

**Comment by lostinplace** [comment]
That's unfortunate, I'm not sure how I feel about open-source projects that are partially closed-source.  I don't think I'll be reasonably able to evaluate the output of your tool every time it's run.

It's a great idea for a project though!

**Comment by nxrighthere** [comment]
Thanks. I'm not excepting any contribution from 3rd-parties, it's not really a problem for me.
